indico/modules/events/contributions/controllers.py

```diff
@@ -107,17 +122,24 @@ def _process(self):
         if form.validate_on_submit():
             contrib = create_contribution(self.event_new, form.data)
             flash(_("Contribution '{}' created successfully").format(contrib.title), 'success')
-            return jsonify_data(html=_render_contribution_list(self.event_new))
+            flash(_("Note that the contribution '{}' might not be displayed in the list due to the enabled filters")
+                  .format(contrib.title), 'info')
+            contrib_report_config = reporter.get_report_config(self.event_new.id)
+            return jsonify_data(**reporter.render_contrib_report(self.event_new, contrib_report_config['filters']))
         return jsonify_form(form)
 
 
 class RHEditContribution(RHManageContributionBase):
     def _process(self):
         form = ContributionForm(obj=FormDefaults(self.contrib), event=self.event_new)
+        form.type.query = self.event_new.contribution_types
         if form.validate_on_submit():
             update_contribution(self.contrib, form.data)
             flash(_("Contribution '{}' successfully updated").format(self.contrib.title), 'success')
-            return jsonify_data(html=_render_contribution_list(self.event_new))
+            flash(_("Note that the contribution '{}' might not be displayed in the list due to the enabled filters")
```

Comment:
Yeah, I need to think of a way to see if the new record matches with the enabled filters.